Added FrameTimings to the useFrameCallback hook (#3455)
software-mansion/react-native-reanimated: docs/docs/api/hooks/useFrameCallback.md

This hook allows you to run a piece of code on every frame update.

```js
useFrameCallback(callback: (frameTimings: FrameTimings) => void, autostart = true): [FrameCallback]
```

### Arguments

#### `callback` [Function]

> Single worklet function that will be called on every frame update.

Reviewer: ```suggestion
A single worklet function that will be called on every frame update.
```

> This function recieves a [`FrameInfo`](#frameinfo-object) object as na argument.

Reviewer: ```suggestion
This function receives a [`FrameInfo`](#frameinfo-object) object as an argument.
```

This hook allows you to run a piece of code on every frame update.

```js
useFrameCallback(callback: (frameTimings: FrameTime) => void, autostart = true): [FrameCallback]
```

### Arguments

#### `callback` [Function]

A single worklet function that will be called on every frame update.

> This function recieves a [`FrameTime`](#frametime-object) object as na argument.

Reviewer: ```suggestion
This function receives a [`FrameTime`](#frametime-object) object as na argument.
```

#### `callback` [Function]

A single worklet function that will be called on every frame update.

> This function receives a [`FrameTime`](#frametime-object) object as na argument.

Reviewer: ```suggestion
This function receives a [`FrameTime`](#frametime-object) object as an argument.
```

Author: This line better be perfect, after all these suggestions.

                    or not (`false`)
* `callbackId: number`: a unique identifier of the callback function

#### `FrameTimings: [object]`

Properties:
* `timestamp: number`: the current system time
* `frameTime: number`: time since last frame - this value may be zero on the first frame update

> * `elapsedTime: number`: time since the callback started running

Reviewer: 1. Add info about unit (in seconds or in milliseconds?) in each line for better readability.
2. What do you think about renaming `frameTime` to `timeSinceLastFrame`?
3. I'm wondering if instead of `frameTime = 0` we can have `isFirstFrame = true`.

>   will be null on the first frame after activation. Starting from the second frame,

Reviewer: ```suggestion
  will be `null` on the first frame after activation. Starting from the second frame,
```

>   it should be ~16ms on 60Hz or ~8ms on 120Hz displays (when there is no lag)

Reviewer: ```suggestion
  it should be ~16 ms on 60 Hz or ~8 ms on 120 Hz displays (when there is no lag)
```

                    or not (`false`)
* `callbackId: number`: a unique identifier of the callback function

#### `FrameInfo: [object]`

Properties:
* `timestamp: number`: the system time (in milliseconds) when the last
  frame was rendered

> * `duration: number | undefined`: time (in milliseconds) since last frame - this value

Reviewer: ```suggestion
* `duration: number | undefined`: time (in milliseconds) since last frame. This value
```

> * `duration: number | undefined`: time (in milliseconds) since last frame - this value
>   will be `undefined` on the first frame after activation. Starting from the second frame,

Reviewer: I think we should use `null` here instead of `undefined`. What do you think?
```suggestion
* `timeSinceLastFrame: number | null`: time (in milliseconds) since last frame - this value
  will be `null` on the first frame after activation. Starting from the second frame,
```

Properties:
* `timestamp: number`: the current system time (in miliseconds)

> * `timeSinceLastFrame: number | null`: time (in miliseconds) since last frame - this value

Reviewer: milliseconds

>   will be null on the first frame after activation (then it should be ~16ms on 60Hz
>   or ~8ms on 120Hz displays when there is no lag)

Reviewer: ```suggestion
  will be null on the first frame after activation. Starting from second frame, it should be ~16 ms on 60 Hz
  or ~8 on 120 Hz displays (when there is no lag)
```

export default function FrameCallbackExample() {
  const x = useSharedValue(0);

  const frameCallback = useFrameCallback((frameTimigs) => {

>     console.log(frameTimings.frameTime + 'ms', 'have passed since the previous frame');

Reviewer: ```suggestion
    console.log(`${frameTimings.frameTime} ms have passed since the previous frame`);
```

export default function FrameCallbackExample() {
  const x = useSharedValue(0);

  const frameCallback = useFrameCallback((frameInfo) => {

>     if (frameInfo.timeSincePreviousFrame == null) {

Reviewer: ```suggestion
    if (frameInfo.timeSincePreviousFrame === null) {
```

export default function FrameCallbackExample() {
  const x = useSharedValue(0);

  const frameCallback = useFrameCallback((frameInfo) => {

>     console.log(`${frameInfo.lastFrameDuration} ms have passed since the previous frame`);

Reviewer: What about the first frame?
